akka-actor/src/main/scala/akka/io/dns/internal/TcpDnsClient.scala

```diff
@@ -38,41 +41,50 @@ import akka.util.ByteString
     case _: Tcp.Connected =>
       log.debug("Connected to TCP address [{}]", ns)
       val connection = sender()
-      context.become(ready(connection))
+      writer = new Writing(connection, log)
+      context.become(ready())
```

Comment:
Mixing a mutable field with the become-style already present here is not great for readability. Let's use just context become closing over state instead.